website/docs/philosophy.md

```diff
@@ -2,20 +2,146 @@
+
+Instead, you have to trigger event from the view itself !
+
+![Statefull component](/img/component_state.gif)
+
+For example, if you want testing the different styling of your like button :
+
+```javascript
+describe('like button style', () => {
+  it('should set button to notLiked by default', () => {
+    const component = shallow(Component, {props: {nbLikes: 1}});
+
+    component.querySelector('button').dispatchEvent('click');
+
+    expect(component.html().querySelector('button').props.class).not.toContain(
+      'liked'
+    );
+  });
+
+  it('should set button to liked when clicking on the button', () => {
+    const component = shallow(Component, {props: {nbLikes: 1}});
+
+    component.querySelector('button').dispatchEvent('click');
+
+    expect(component.html().querySelector('button').props.class).not.toContain(
+      'liked'
+    );
+  });
+});
+```
+
+### Component can emit event
+
+**Important: As only react is supported right now, theses feature are not yet implemented because react don't need it**
+
+As component
+
+![Component that emit event](/img/component_event.png)
+
+```javascript
+describe('like button - onLike event', () => {
+  it('should emit onClick Event', () => {
+    const spy = createSpy();
+    // const spy = jest.fn(); on jest
+    const component = shallow(Component, {events: {onLike}});
+
+    component.querySelector('button').dispatchEvent('click');
+
+    expect(spy).toHaveBeenCalled();
+  });
+});
+```
+
+### Component sometimes have more than props
+
+**Important: As only react is supported right now, theses feature are not yet implemented because react don't need it**
+
+For example, Angular component can inject services in the constructor of the components.
+In order to provide a standardized way external object that can affect the rendered view other than props, a external option is available.
+
+![Component with externals](/img/component_externals.png)
+
+An example :
+
+```javascript
+describe('user component', () => {
+  it('should display the user got from the service', () => {
+    const component = shallow(Component, {
+      externals: {
+        userService: {
+          getUser: () => Promise.resolve({name: 'component-test-utils'})
+        }
+      }
+    });
+
+    expect(component).toContain('component-test-utils');
+  });
+});
+```
+
+### Mocking strategies
+
+Component doesn't only generate HTML element, they can have sub-components in their view. For this kind of "parent" component, the main question of the developer should be: "should I mock this child component ?".
+
+For this use case, Component-test-utils have two way to create a component in an test environment:
 
-To have more information on testing, you can check these slides (in french) : https://slides.com/florentberthelot/test-composants-web#/
+- _White list_: Every sub-component is mocked, you can give a list a component that won't be mocked and give their mock.
```

Comment:
```suggestion
- _White list_: Every sub-component is mocked automatically and you can specify the mock for each component by providing a map.
```

Reply:
I hope I understood correctly what you meant.
Maybe an example could be useful here, at least a philosophical one 💡